From c2705a51e77a69ae36f31e9c34a73e4e58ea45c2 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Fri, 2 Feb 2018 11:02:27 +0200
Subject: [PATCH 3408/5242] MLK-17485: ASoC: fsl_sai: Specify supported
 rate_min and rate_max

commit  6694d5ebe439790c3a33bc7cc363732daf0f0c69 from
https://source.codeaurora.org/external/imx/linux-imx.git

Because fsl_sai_dai rates doesn't have a specific set of
rate values (.rates = SNDRV_PCM_RATE_KNOT) we need to provide
rate_min and rate_max otherwise functions trying to get
supported parameters will get confused and return an error.

Fixes:  1b6f0496e013 ("MLK-17428-8: ASoC: fsl_sai: support 768KHz sample rate")
Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_sai.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index 52a2ef4..7772998 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -928,6 +928,8 @@ static int fsl_sai_dai_resume(struct snd_soc_dai *cpu_dai)
 		.stream_name = "CPU-Playback",
 		.channels_min = 1,
 		.channels_max = 32,
+		.rate_min = 8000,
+		.rate_max = 2822400,
 		.rates = SNDRV_PCM_RATE_KNOT,
 		.formats = FSL_SAI_FORMATS,
 	},
@@ -935,6 +937,8 @@ static int fsl_sai_dai_resume(struct snd_soc_dai *cpu_dai)
 		.stream_name = "CPU-Capture",
 		.channels_min = 1,
 		.channels_max = 32,
+		.rate_min = 8000,
+		.rate_max = 2822400,
 		.rates = SNDRV_PCM_RATE_KNOT,
 		.formats = FSL_SAI_FORMATS,
 	},
-- 
1.7.9.5

